File: exercicios/Python/1983/1983.py
def entrada():
    n = int(input())
    return(n)

def main():
    n = entrada()
    ArrAluno = []
    ArrNota=[]
    cont = 0
    aux = 0
    pos = 0

    for i in range(n):
        aluno, nota = input().split()
        aluno = int(aluno)
        nota = float(nota)

        if nota >= 8:
            ArrAluno.append(int(aluno))
            ArrNota.append(float(nota))
    
    for x in ArrAluno:
        if x > aux:
            aux = x
            pos = x
    if pos == 0:
        print('Minimum note not reached')
    else:
        print(pos)
main()

File: exercicios/JavaScript/1983/1983.js
var input = require('fs').readFileSync('stdin','utf8');
var lines = input.split('\n');

var x = parseInt(lines.shift())

var notaMax = 0;
var matMax = 0;

for(let i=0; i<x;i++){
    var v = lines.shift().split(' ');
    matricula = v[0];
    nota = parseFloat(v[1]).toFixed(2)

    if(nota > 8){
        if(nota > notaMax){
            notaMax = nota
            matMax = matricula
        }
    }
}

if(matMax > 0){
    console.log(matMax)
}else{
    console.log("Minimum note not reached");
}
